hazelcast/src/main/java/com/hazelcast/internal/memory/impl/ByteBufferMemoryAccessor.java

```diff
@@ -0,0 +1,396 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.memory.impl;
+
+
+import com.hazelcast.internal.memory.MemoryAccessor;
+import com.hazelcast.util.Preconditions;
+
+import java.lang.reflect.Field;
+import java.nio.ByteOrder;
+
+public class ByteBufferMemoryAccessor implements MemoryAccessor {
+    private static final boolean BIG_ENDIAN = ByteOrder.nativeOrder() == ByteOrder.BIG_ENDIAN;
+
+    private void validateObject(Object srcObj) {
+        Preconditions.checkNotNull(srcObj);
+        Preconditions.checkTrue(srcObj instanceof byte[], "object should be byte-array");
+    }
+
+    private void checkOffset(long offset) {
+        if (offset < Integer.MIN_VALUE || offset > Integer.MAX_VALUE) {
```

Comment:
can `offset` be negative? shouldn't `offset < Integer.MIN_VALUE` be `offset < 0`?